include/mruby/error.h

```diff
@@ -132,6 +132,14 @@ MRB_API mrb_value mrb_rescue_exceptions(mrb_state *mrb, mrb_func_t body, mrb_val
                                         mrb_func_t rescue, mrb_value r_data,
                                         mrb_int len, struct RClass **classes);
 
+typedef mrb_value mrb_vm_protect_func(mrb_state *mrb, void *userdata);
+
+/**
+ * This API function behaves like `mrb_protect()`.
+ * The advantage is that it avoids objectifying the user data.
+ */
+MRB_API mrb_value mrb_vm_protect(mrb_state *mrb, mrb_vm_protect_func *body, void *userdata, mrb_bool *error);
```

Comment:
The `_vm` functions are to be called from VM, but this function is not. We should rename the function.

Reply:
Oh, I see. I will think of another name.

Reply:
Thank you. I am sorry but I am not 100% satisfied with the new name either. But I will merge this anyway, and rename the function later.